### PR TITLE
Completed week 3 assessment.

### PR DESCRIPTION
**Pull request by ghost**
Submitting pull request for week 3 assessment.

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,42 +15,118 @@
...
 // Example input: 6
 // Expected output: [1, 1, 2, 3, 5, 8]
+describe("fib",()
+    it("takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence", () => {
+
+      //an expect method, nested within the test block, calling on the hello() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+      expect(fib(num1)).toEqual([1,1,2,3,5,8])
+      expect(fib(num2)).toEqual([1,1,2,3,5,8, 13, 21, 34, 55])
+    })
+  })
```
Comment (elysemontano) [inline code-challenges.test.js:35]: The layout for your test looks great.  There are a few small syntax errors and I don't see the variables num1 and num2 defined anywhere.  Also, make sure to run the tests and see what errors you get back.

File: code-challenges.test.js [hunk 2]
```diff
@@ -15,42 +15,118 @@
...
+
+// Create a function that declare fib as a variable in an array
+// Use the array to store the Fibonacci to eliminate the need to use auxiliar variables in the code
+// Input 10 following .notation in order to render the first 10 numbers of the Fibonacci output array
+
+var fib = [1, 1, 2];
+for(var i=fib.length; i<10; i++) {
+    fib[i] = fib[i-2] + fib[i-1];
+}
+console.log(fib);
```
Comment (elysemontano) [inline code-challenges.test.js]: Your logic looks good and this definitely works for this instance.  Currently, this is not a function making it so you need to code out each instance you want to check.  Your test looks like it is looking for a function called fib which doesn't exist yet.  When possible, if you find yourself repeating code, try to refactor to turn it into a function.

File: code-challenges.test.js [hunk 3]
```diff
@@ -15,42 +15,118 @@
...
+  it("takes in an array and returns a new array of only odd numbers sorted from least to greatest", () => {
+    const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+    // Expected output: [-9, 7, 9, 199]
+    const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
+    expect(oddNumbersInOrder(fullArr1)).toEqual([-9, 7, 9, 199])
+    expect(oddNumbersInOrder(fullArr2)).toEqual([-823, 7, 23])
+  })
+})
```
Comment (elysemontano) [inline code-challenges.test.js:73]: This looks excellent!

File: code-challenges.test.js [hunk 4]
```diff
@@ -15,42 +15,118 @@
...
+// filter every value that isn't a number and every number that is not odd
+// store filtered values on new variable
+// sort the final results
+
+const oddsOnly = (arr) => {
+    let oddNumbers = arr.filter(n => {
+        return n % 2 !== 0 && typeof n === "number"
+    })
+    return oddNumbers.sort((a,b) => a - b)
+}
```
Comment (elysemontano) [inline code-challenges.test.js:91]: Great!  Just make sure that your function name written here matches the name in your test.

File: code-challenges.test.js [hunk 5]
```diff
@@ -15,42 +15,118 @@
...
 
-const numbersToAdd3 = []
+    const numbersToAdd3 = []
 // Expected output: []
-
+    expect(sum(numbersToAdd1)).toEqual([2, 6, 51, 60])
+    expect(sum(numbersToAdd2)).toEqual([0, 7, -1, 11])
+    expect(sum(numbersToAdd3)).toEqual([])
+  })
+})
```
Comment (elysemontano) [inline code-challenges.test.js:113]: Excellent work!

File: code-challenges.test.js [hunk 6]
```diff
@@ -15,42 +15,118 @@
...
+// declare the empty array newArr
+// return value of index zero as condtional
+// return the new array of values each containing an accumelating sum
+
+const sumOfArray = (arr) => {
+    let arrAccumulate = 0
+    return arr.map(n => {
+        return arrAccumulate += n
+    })
+}
```
Comment (elysemontano) [inline code-challenges.test.js:133]: This looks great!  Just make sure that you match your function name to your test calls as well.

File: interview-questions.md [hunk 7]
```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: The keyword 'this' in JS refers to an object.
 
-  Researched answer:
+  Researched answer: In JavaScript, the this keyword refers to an object. Which object depends on how this is being invoked (used or called). The this keyword also refers to different objects depending on how it is used:. In an object method, this refers to the global object.
```
Comment (elysemontano) [inline interview-questions.md]: Researched answers should be formed in your own words.

File: react-debugging.html [hunk 8]
```diff
@@ -79,10 +80,10 @@ <h1>Draw a Card</h1>
                   )
                 })}
               </div>
-          )
+              </>
```
Comment (elysemontano) [inline react-debugging.html:83]: Great job on adding react fragments and adding the draw method onto the onClick.  If you are still seeing an error, "this" will be needed when calling the methods.

**Review by elysemontano — CHANGES_REQUESTED** [review]
Hi Nico, I am requesting some changes on this assessment.  Please make sure to go back and update your researched answers to be in your own words.  Your logic in the code challenges is looking great.

**Comment by ghost** [comment]
Made requested changes.

**Review by elysemontano — APPROVED** [review]
Great!  Thank you for making those changes.

**Pull request merged**